### DISCUSSION TITLE
Built in clipboard that allows direct upload to websites

### DISCUSSION THREAD

**Discussion by gyithub** [Floorp-Projects/Floorp#822] [Feature Request]
**Inbuilt Clipboard:**
So, OperaGX has this inbuilt clipboard that allows you access your recent copied attachments(images etc) and files you have downloaded recently via a popup when you use it, what makes it different from native `ctrl+c` &` ctrl+v` is that this clipboard allows you to input into websites _**where native copy paste isn't allowed**_, you have to first save the image then upload it to the website.

For example: lets consider using the website [remove.bg](https://www.remove.bg/upload)

when I click on `Upload Image` in OperaGX I get:
![image_2024-01-31_175401432](https://github.com/Floorp-Projects/Floorp/assets/158288047/9fa0a99f-a0a8-4447-a02e-2a763e1d9d27)
_(Left is my most recent clipboard image, and the right is the most recent downloads and If i click show all files, it opens the file explorer)_

Now, when I try the same website on Floorp, I get a file explorer window asking me to select an image to upload, so basically to do the same I have to first download henrycavill.jpg then upload it.

This might seem like a minor issue/ nice to have but it is just so useful, anytime you want to upload something you can easily access recent files. say you are trying to compress a video using those online compressors, its so easy.

In fact to make this very request, I haven't downloaded any image, opera allowed me to directly paste the attachment from my clipboard.

Would be amazing if this feature is added

**Comment by Telmesomething** [comment]
Opera is a Chrome based browser and behaves differently to Floorp which is open source. Be aware that everything you do on Opera or any other Chrome based browser is being recorded and tracked.

**Reply by gyithub** [reply]
I am aware about the controversies regarding OperaGX, I made a [post](https://www.reddit.com/r/Floorp/comments/1afa05c/feature_recommendation_i_would_love_if_floorp_had/) regarding this feature in the floorp subreddit and it gathered some interest, I really like this feature of OperaGX and wanted to know if it could be implemented in Floorp as I have now fully switched to Floorp for the time being, so I created this request.
